subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1655,20 +1819,35 @@ public List<String> getResolutionAlternatives() {
     }
 
     @Override
-    public DeprecatableConfiguration deprecateForDeclaration(String... alternativesForDeclaring) {
+    public DeprecatableConfiguration deprecateForDeclarationAgainst(String... alternativesForDeclaring) {
+        validateMutation(MutationType.USAGE);
         this.declarationAlternatives = ImmutableList.copyOf(alternativesForDeclaring);
```

Comment:
It would be annoying to update all those tests, but I'm fine with doing it - if we really do want to remove those methods.

Reply:
Since the `deprecateFor` methods are both existing and internal API I'm fine with leaving them around for now. I don't really want to have to add many more changes to this PR anyways

Reply:
I mostly just wanted to point out how we never actually call these methods with parameters (in production code)